src/Bridge/Doctrine/EventListener/PublishMercureUpdatesListener.php

```diff
@@ -259,12 +260,6 @@ private function getGraphQlSubscriptionUpdates($object, array $options, string $
      */
     private function buildUpdate($iri, string $data, array $options): Update
     {
-        if (method_exists(Update::class, 'isPrivate')) {
-            return new Update($iri, $data, $options['private'] ?? false, $options['id'] ?? null, $options['type'] ?? null, $options['retry'] ?? null);
-        }
-
-        // Mercure Component < 0.4.
-        /* @phpstan-ignore-next-line */
-        return new Update($iri, $data, $options);
+        return new Update($iri, $data, $options['private'] ?? false, $options['id'] ?? null, $options['type'] ?? null, $options['retry'] ?? null);
```

Comment:
Why removing the BC layer?

Reply:
this PR drops support for older mercure/mercure-bundle versions. do you think we should keep them? 🤔

Reply:
We can drop it if it's really necessary but we should add a conflict then.

Reply:
indeed this is the main issue, as API Platform is a library we don't have any `compose.lock` and therefore users may have issues installing api platform with an incompatible mercure-bundle

Reply:
I think the best choice would be to add a BC layer, and drop it in the next major ( however, we should add a conflict with mercure-bundle > 0.5 for future releases, otherwise we would end up with a pile of BC layers to deal with ).

Reply:
I'm :+1: for adding a conflict with mercure-bundle < 0.6 for `2.7` (it needs to be documented in the CHANGELOG).

Reply:
The last version of the Mercure bundle is the 0.2.6, isn't it? I'm lost :sweat_smile:

Reply:
yea 😆 i meant to say 0.3, the last version of mercure component is 0.4, next ( the one compatible with mercure-bundle 0.3 ) is 0.5

Reply:
I think we should at least have a BC layer for the previous version (so in this case the `0.2` version of Mercure bundle) to allow a smooth upgrade path.

Reply:
make sense 👍